middleware.go

```diff
@@ -10,21 +10,9 @@ import (
 	"github.com/0xsequence/authcontrol/proto"
 )
 
-type Options struct {
-	KeyFuncs   []KeyFunc
-	UserStore  UserStore
-	ErrHandler func(r *http.Request, w http.ResponseWriter, err error)
-}
-
-func Session(auth *jwtauth.JWTAuth, o *Options) func(next http.Handler) http.Handler {
-	eh := defaultErrHandler
-	if o != nil && o.ErrHandler != nil {
-		eh = o.ErrHandler
-	}
-
-	var keyFuncs []KeyFunc
-	if o != nil {
-		keyFuncs = o.KeyFuncs
+func Session(auth *jwtauth.JWTAuth, u UserStore, eh ErrHandler, keyFuncs ...KeyFunc) func(next http.Handler) http.Handler {
```

Comment:
Please son't use variadic parametrs. I would keep the options struct. It is easier to reas and better for preventing braking changes. It is also more readable.